### PR TITLE
enable function description

### LINKED ISSUE #1367: Enable function.description
### What is the expected enhancement?
The `description` of `function` is not used now.
Enable the `description` for the function user to look up the function details

The function user can do to see the description of the function like:
```
func_a = client.get("function_a")
func_a.description
```

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
fix #1367

### CODE REVIEW

**Review by psschwei — APPROVED** [review]

File: docs/deployment/deploying_custom_image_function.rst [hunk 1]
```diff
@@ -183,16 +183,47 @@ Upload and register function
...
+        observable: observable 
+    """
 
     function_with_custom_image = QiskitFunction(
         title="custom-image-function",
         image="<image retistory/image name:image tag>",
-        provider="<provider id>"
-    )
+        provider="<provider id>",
+        description=help
```
Comment (Tansito) [inline docs/deployment/deploying_custom_image_function.rst:200]: I would improve a little bit this string maybe with a similar format than the previous ones:

```
description="# markdown tex"
```

File: gateway/api/models.py [hunk 2]
```diff
@@ -86,6 +86,7 @@ class Program(ExportModelOperationsMixin("program"), models.Model):
     created = models.DateTimeField(auto_now_add=True, editable=False)
 
     title = models.CharField(max_length=255, db_index=True)
+    description = models.TextField(null=True, blank=True)
```
[the remark this answers is not in the document]
Comment (Tansito) [reply gateway/api/models.py:89]: Oh ok, we had it only in `QiskitFunction` client entity!

**Review by Tansito — COMMENTED** [review]
Can we update `upload` and `list` tests checking that we are storing the new field and returning it correctly?

File: docs/deployment/deploying_custom_image_function.rst (hunk 1 above)
Comment (Tansito) [reply docs/deployment/deploying_custom_image_function.rst:200]: Aaaah ok, I thought it was just the `"help"` string hahaha, got it

**Comment by akihikokuroda** [comment]
@Tansito I added a small test.

**Review by Tansito — APPROVED** [review]
Thank you, @akihikokuroda 🙏 ! LGTM

**Pull request merged**